### PR TITLE
LOG-1160: Added credentials update event

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### Description
CLO will generate an event after updating certificates

/cc @jcantrill 
/assign @jcantrill 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-1160

### CODE REVIEW

**Review by jcantrill — COMMENTED** [review]
/hold
/approve

@igor-karpukhin  please make sure we alter "must-gather" to gather events from the namespace if it doesn't already do so

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ const (
 	LegacySyslog        = "_LEGACY_SYSLOG"
 
 	FluentdImageEnvVar = "FLUENTD_IMAGE"
+	CertEventName      = "cluster-logging-certs-generate"
```
[the remark this answers is not in the document]
Comment (igor-karpukhin) [reply pkg/constants/constants.go:38]: This is the name of the event. Do we want to use `openshift-logging` for all events?

File: pkg/controller/clusterlogging/clusterlogging_controller.go [hunk 2]
```diff
@@ -86,7 +92,7 @@ func (r *ReconcileClusterLogging) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
-	if err = k8shandler.Reconcile(instance, r.client); err != nil {
+	if err = k8shandler.Reconcile(instance, r.client, r.recorder); err != nil {
```
[the remark this answers is not in the document]
Comment (igor-karpukhin) [reply pkg/controller/clusterlogging/clusterlogging_controller.go:95]: I'd say we could just pass `ClusterLoggingRequest`. So, build it before the `Reconcile` function, not inside it: `clusterLoggingRequest := ClusterLoggingRequest{
		Client:        requestClient,
		Cluster:       requestCluster,
		EventRecorder: r,
	}
`

**Comment by igor-karpukhin** [comment]
@jcantrill events are emitted against k8s resources. So this event will be visible in the `clusterlogging` instance object in the events section. No need to modify `must-gather`

File: pkg/constants/constants.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcantrill) [reply pkg/constants/constants.go:38]: i was asking if we should prefer something like "openshift-logging-certs-generate" or maybe it should just be "openshift-logging-certs"

File: pkg/k8shandler/certificates.go [hunk 3]
```diff
@@ -98,6 +98,8 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCertificates() (err e
 			}
 		}
 
+		clusterRequest.EventRecorder.Event(clusterRequest.Cluster, "Normal", "Updated", "certificates successfully updated")
```
[the remark this answers is not in the document]
Comment (igor-karpukhin) [reply pkg/k8shandler/certificates.go]: Fixed by parsing the cert generation output

**Comment by igor-karpukhin** [comment]
@jcantrill The result looks like this:
````
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/ca.crt reason FileMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/ca.key reason FileMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/ca.crt reason ExpiredOrMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [reason Missing type File intialization msg /tmp/ocp-clo/ca.db]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/ca.serial.txt reason Missing type File intialization]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/system.logging.fluentd.crt reason FileMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/system.logging.fluentd.crt reason ExpiredOrMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [msg /tmp/ocp-clo/system.logging.kibana.crt reason FileMissing type Regenerate]
  Normal  Updated  9s               clusterlogging-controller  [reason ExpiredOrMissing type Regenerate msg /tmp/ocp-clo/system.logging.kibana.crt]reason ExpiredOrMissing type Regenerate]

````

**Comment by jcantrill** [comment]
cc @alanconway @ewolinetz

File: pkg/k8shandler/certificates.go [hunk 4]
```diff
@@ -97,6 +98,14 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCertificates() (err e
 				return err
 			}
 		}
+		for _, r := range result {
+			strs := r.([]interface{})
+			for _, s := range strs {
+				if s == "reason" {
+					clusterRequest.EventRecorder.Event(clusterRequest.Cluster, "Normal", "Updated", fmt.Sprint(strs))
```
Comment (jcantrill) [inline pkg/k8shandler/certificates.go:105]: Is there a potential for an error here that we should at least capture and log?

File: pkg/constants/constants.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcantrill) [reply pkg/constants/constants.go:38]: @alanconway any opinions on the prefix here?

**Comment by jcantrill** [comment]
@igor-karpukhin do we need to update the must-gather or do we already get these events?

**Comment by igor-karpukhin** [comment]
@jcantrill I thought that events are collected already by collecting `clusterlogging` instance. But that didn't work. Fixed by explicitly collecting `clusterlogging` events.

File: pkg/constants/constants.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alanconway) [reply pkg/constants/constants.go:38]: Looking at OLM events, they just use the namespace name plus a unique identifier to construct event names, e.g.
`event/redhat-operators.166f0764288aa074`
Events names are not very significant, `oc get events` and the console events panel don't even show the name. The interesting fields are the reason, message and related-object. So I'd say all events we generate in the openshift-logging namespace  just be called openshift-logging.<unique-id>

File: pkg/k8shandler/certificates.go (hunk 4 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/certificates.go:105]: Function `Event` doesn't return any value or errors. But, I think I missed the interface conversion check at line 102

**Pull request merged**